elastic/beats
# Add memory/cgroup fields to metricbeat Kibana module

This PR updates the Metricbeat `kibana` module with support for the new memory and cgroup fields added to Kibana in https://github.com/elastic/kibana/pull/172146. We plan to use this metric for serverless monitoring, so metricbeat needs to know about it.

## Checklist


## How to test this PR locally

- run kibana from source & ES from snapshot as explained in https://www.elastic.co/guide/en/kibana/master/development-getting-started.html
- pull down this PR, `cd metricbeat`
- `mage build`
- `./metricbeat modules enable kibana`
- uncomment the `elasticsearch.output` username/password lines in `metricbeat.yml` (`elastic:changeme`)
- update `./modules.d/kibana.yml`:
```yml
- module: kibana
  metricsets:
    - stats # be sure to enable stats metricset
  period: 10s
  hosts: ["localhost:5601"]
  basepath: "abc" # whatever your base path is
  username: "elastic"
  password: "changeme"
```
- `./metricbeat -e`
- create `metricbeat-*` data view in kibana
- you should now be seeing some log entries which contain the new fields, eg. `process.memory.array_buffers`

<img width="962" alt="Screenshot 2023-11-30 at 13 29 41" src="https://github.com/elastic/beats/assets/1532934/654b461c-6afb-4d04-9364-e353044c7644">
